src/SearchByTitleForm/SearchByTitleForm.js

```diff
@@ -19,32 +21,50 @@ class SearchByNameForm extends Component {
     }
 
     handelSubmit = (event) => {
-        event.preventDefault()
-        const movie = this.props.findMovieByTitle(this.state.movieTitle);
-        this.resetMovieTitle();
+        event.preventDefault();
+        const id = this.props.findMovieByTitle(this.state.movieTitle);
+        if(id) {
+            this.props.history.push('/' + id);
```

Comment:
This is the line that pushes into history. The `withRouter` gives access to the history object, which is a stack of all the pages we have been on. By pushing into it we change the current page we are on, when the page is changed to one of the movie's id's the Movie component uses that path to fetch the movie.

Reply:
whoa this is a really cool solution!